packages/array-index/index.tests.ts

```diff
@@ -0,0 +1,15 @@
+import index = require('./index');
+
+// OK
+index([ { a: 5 }, { a: 6 } ], 'a');
+index([ { a: 5 }, { a: 6 }, null, undefined ], 'a');
+
+// Not OK
+// @ts-expect-error
+index([ { a: 5 }, { a: 6 } ], 'b');
+// @ts-expect-error
+index([ { a: 5 }, { a: 6 } ], (o: number) => o.a);
```

Comment:
I'd prefer not to use typescript in the test itself
So just
`index([ { a: 5 }, { a: 6 } ], (o) => o.a);`